currencyii/src/main/java/nl/tudelft/trustchain/currencyii/util/DAOJoinHelper.kt

```diff
@@ -212,17 +217,31 @@ class DAOJoinHelper {
                 walletManager.protocolECKey()
             )
             val signatureSerialized = signature.encodeToDER().toHex()
-            val agreementData = SWResponseSignatureTransactionData(
-                blockData.SW_UNIQUE_ID,
-                blockData.SW_UNIQUE_PROPOSAL_ID,
-                signatureSerialized
-            )
-
-            trustchain.createProposalBlock(
-                agreementData.getTransactionData(),
-                myPublicKey,
-                agreementData.blockType
-            )
+            if (votedInFavor) {
+                val agreementData = SWResponseSignatureTransactionData(
+                    blockData.SW_UNIQUE_ID,
+                    blockData.SW_UNIQUE_PROPOSAL_ID,
+                    signatureSerialized
+                )
+
+                trustchain.createProposalBlock(
+                    agreementData.getTransactionData(),
+                    myPublicKey,
+                    agreementData.blockType
+                )
+            } else {
+                val negativeResponseData = SWResponseNegativeSignatureTransactionData(
```

Comment:
No, I don't think so, what if the constructor is used in the future for other purposes where not such blockData is present. I think this is the cleanest way, but convince me otherwise haha